sdk-java/src/main/java/com/spotify/confidence/FlagResolverClientImpl.java

```diff
@@ -3,18 +3,30 @@
 import com.google.protobuf.Struct;
 import com.google.protobuf.Value;
 import com.spotify.confidence.shaded.flags.resolver.v1.ResolveFlagsResponse;
+import com.spotify.confidence.telemetry.Telemetry;
+import java.time.Duration;
+import java.time.Instant;
 import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nullable;
 
 class FlagResolverClientImpl implements FlagResolverClient {
   public static final String OPEN_FEATURE_RESOLVE_CONTEXT_KEY = "open-feature";
   private final FlagResolver grpcFlagResolver;
+  private final Telemetry telemetry;
 
-  public FlagResolverClientImpl(FlagResolver grpcFlagResolver) {
+  public FlagResolverClientImpl(FlagResolver grpcFlagResolver, @Nullable Telemetry telemetry) {
```

Comment:
maybe a nit: should we pass a noop Telemetry rather than a null object here? Thus avoiding null checks later in this class

Reply:
Actually no. I tried it out and I didn't like the result.